pkg/download/inmemory/inmemory.go

```diff
@@ -0,0 +1,45 @@
+package inmemory
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/cavaliergopher/grab/v3"
+)
+
+// Download takes an endpoint in the form of a string (URL) or a *http.Request
+// allowing caller to pass a custom http request to the downloader library, it
+// takes a http.RoundTripper as the second argument to be used by the client
+func Download[T string | *http.Request](endpoint T, transport http.RoundTripper) (io.ReadCloser, error) {
```

Comment:
You can avoid the use of the generics by always setting the user-agent in this method, which makes sense in my opinion.

Reply:
i think in zabbix we ignore requests with http agent `crc *` from the download data count, with the current use of this (for getting release info and bundle hashes) it should be okay, will remove the type parameters and set the user-agent in the function itself

Reply:
more import cycles, to set the user agent to `crc/<crc_version>`we have to import the `version` package in `inmemory` package and the version package has to import the `inmemory` package to download the `release-info.json`

Reply:
We still want to be consistent imo and always use the same user agent. I guess for now you could add a `userAgent string` parameter to `Download`, and a `version.UserAgent()`, and pass it explicitly. Otherwise this needs more code movement.

Reply:
does it functionally change anything? what are you planning to do with the 'useragent'; track it as telemetry?

Reply:
Downloading the crc version file is done with user agent set, downloading the signature file is done without a user agent set, and there are 2 code paths in the download code to deal with both situations.
I don't mind dropping the user agent or always setting it, but it would be better to always do the same thing.

See also https://github.com/crc-org/crc/pull/3693#discussion_r1213949744

Reply:
`User-Agent` seems is not used to filter out http requests from `crc`, not sure how i reached that, but it doesn't make sense also as for collecting the download data from splunk we can use a query language and can filter out the `release-info.json` or the `sha256sum.txt.sig` files.